core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -317,6 +322,77 @@ public String toString() {
   }
 
   /**
+   * Provides arguments for a {@link SubchannelPicker#pickSubchannel(PickSubchannelArgs)}.
+   */
+  public abstract static class PickSubchannelArgs {
```

Comment:
I am thinking about reverting back to `abstract`. The channel would have its own private implementation which will not have a builder, but just a constructor. Unit tests would extend the abstract class to make fake values. Unfortunately Mockito would be discouraged because any unstubbed methods (e.g., new fields) will return null.